packages/kbn-dev-utils/src/vscode_config/managed_config_keys.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+
+import { REPO_ROOT } from '@kbn/utils';
+
+export interface ManagedConfigKey {
+  key: string;
+  value: any;
+}
+
+/**
+ * Defines the keys which we overrite in user's vscode config for the workspace
+ */
+export const MANAGED_CONFIG_KEYS: ManagedConfigKey[] = [
+  {
+    key: 'files.watcherExclude',
```

Comment:
So, in order to support managing the `search.exclude` property it felt like people needed to be able to add exclusions for themselves, so I've updated the logic to define management comments on each property it writes like so:

```
/**
 * @managed
 *
 * Some settings in this file are managed by @kbn/dev-utils. When a setting is managed it is preceeded
 * with a comment "// @managed" comment. Replace that with "// self managed" and the scripts will not
 * touch that value. Put a "// self managed" comment at the top of the file, or above a group of settings
 * to disable management of that entire section.
 */
{
  "workbench.colorTheme": "Noctis Obscuro",
  "files.watcherExclude": {
    // self managed
    "**/.eslintcache": false,
    // @managed
    "**/.es": true,
    // @managed
    "**/.yarn-local-mirror": true,
    // @managed
    "**/.chromium": true,
    // @managed
    "**/bazel-*": true,
    // @managed
    "**/node_modules": true,
    // @managed
    "**/target": true,
    // @managed
    "**/*.log": true,
    // @managed
    "**/packages/kbn-pm/dist/index.js": true
  },
  "search.exclude": {
    // @managed
    "**/packages/kbn-pm/dist/index.js": true
  }
}
```

Now, when the script runs it will ignore any property which doesn't have a `@managed` comment, and delete `@managed` properties which are no longer managed. Additionally, users can self manage a property by replacing the `@managed` comment with `// self managed` and then customize things further.